mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -77,6 +83,79 @@ type providerMeta struct {
 	ModuleName string `cty:"module_name"`
 }
 
+type Formatter struct {
+	TimestampFormat string
+	LogFormat string
+}
+
+// Borrowed logic from https://github.com/sirupsen/logrus/blob/master/json_formatter.go and https://github.com/t-tomalak/logrus-easy-formatter/blob/master/formatter.go
+func (f *Formatter) Format(entry *logrus.Entry) ([]byte, error) {
+	if !logging.IsDebugOrHigher() {
+		return nil, nil // Suppress logs if TF_LOG is not DEBUG or TRACE
+	}
+	output := f.LogFormat
+	entry.Level = logrus.DebugLevel // Force Entries to be Debug
```

Comment:
Do you believe it's worth switching to `TraceLevel` for when TF_LOG is `TRACE` ?

Reply:
No, debug is verbose enough. Also, we are stripping out grpc internals log anyway so no benefit there.